src/Driver/PhpAmqpLibDriver.php

```diff
@@ -326,4 +337,25 @@ private static function getReplyTo(AMQPMessage $message)
         return $message->has(self::REPLY_TO) ?
             $message->get(self::REPLY_TO) : '';
     }
+
+    private function retryPublish($exchangeName, Message $message)
+    {
+        $this->numberOfRetry++;
+        $this->connection->reconnect();
+        $this->channel = null;
+        $this->publish($exchangeName, $message);
+        $this->numberOfRetry = 0;
+    }
```

Comment:
Si je suis bien le flow d’execution, `numberOfRetry` est plutôt utilisé comme un flag, plus que comme une valeur absolu. Si on garde la logique du compteur, ça vaudrait peut-être le coup d’avoir une constante `MAX_RETRIES`, afin de le comparé dans le `checkIfWeShouldRetry()` et éviter de le ré-initialiser dans le `retryPublish()`. Si on considère une approche _flag_  on pourrait avoir un _flag_  `retryPublish` à `true` en début de `retryPublish`… Pas sûr qu’on ait besoin de la réinitialisation `$this->numberOfRetry = 0;`.

Reply:
J'ai modifié pour que ça soit un flag, ça correspond plus a ce que je veux faire pour le moment. Et j'ai rajouté un test pour montrer l'importance de la réinitialisation après le retry, dans le cas de perte de connexion consécutive.